code/ui/components/keybind/Keybind.cs

```diff
@@ -0,0 +1,61 @@
+using Sandbox;
+using Sandbox.UI;
+
+namespace TTTReborn.UI
+{
+    public partial class Keybind : TextEntry
+    {
+        public string BoundKey { get; private set; } = "";
+
+        public string BoundCommand { get; set; } = "";
+
+        public Keybind() : base()
+        {
+            StyleSheet.Load("/ui/components/keybind/Keybind.scss");
+
+            AcceptsFocus = true;
+        }
+
+        public override void OnButtonTyped(string button, KeyModifiers km)
+        {
+            base.OnButtonTyped(button, km);
+
+            BoundKey = button;
+            Text = button;
+            CaretPos = button.Length;
+
+            Blur();
+
+            if (!string.IsNullOrEmpty(BoundCommand))
+            {
+                DialogBox dialogBox = new DialogBox(this.FindRootPanel());
+                dialogBox.TitleLabel.Text = $"Bind '{BoundCommand}'";
+                dialogBox.AddText($"Are you sure that you wanna bind '{BoundCommand}' to '{BoundKey}'?");
+                dialogBox.OnAgree = (panel) =>
+                {
+                    ConsoleSystem.Run($"bind {BoundKey} {BoundCommand}");
+
+                    panel.Close();
+                };
+                dialogBox.OnDecline = (panel) => panel.Close();
+                dialogBox.Display();
+            }
+        }
+    }
+}
+
+namespace Sandbox.UI.Construct
+{
+    using TTTReborn.UI;
```

Comment:
Double namespaces?

Reply:
This is needed as we're in a different namespace

Reply:
just making sure : D